index.js

```diff
@@ -1 +1,112 @@
-// implement your API here
+const express = require("express");
+const db = require("./data/db.js");
+const server = express();
+
+server.use(express.json());
+
+server.post("/api/users", (req, res) => {
+    const newUser = req.body;
+    const { name, bio } = req.body;
+
+    if (name && bio) {
+        db.insert(newUser)
+            .then(addedUser => {
+                res.status(201).json(addedUser);
+            })
+            .catch(err => {
+                res.status(500).json({
+                    error:
+                        "There was an error while saving the user to the database"
+                });
+            });
+    } else {
+        res.status(400).json({
+            error: "Please provide name and bio for the user."
+        });
+    }
+});
+
+server.get("/", (req, res) => {
+    res.send("Hello World!");
+});
+
+server.get("/api/users", (req, res) => {
+    db.find().then(allUsers => {
+        if (allUsers) {
+            res.json(allUsers);
+        } else {
+            res.status(500).json({
+                error: "The users information could not be retrieved."
+            });
+        }
+    });
+});
+
+server.get("/api/users/:id", (req, res) => {
+    const { id } = req.params;
+    db.findById(id)
+        .then(foundUser => {
+            if (foundUser) {
+                db.findById(id).then(res.status(201).json(foundUser));
+            } else {
+                res.status(404).json({
+                    error: "The user with the specified ID does not exist."
+                });
+            }
+        })
+        .catch(err => {
+            res.status(404).json({
```

Comment:
I would make your catch statements 500 level errors. Usually when you get there, that means there is something wrong with the server or database. 400 level errors are reserved for client side errors usually.